presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java

```diff
@@ -167,6 +170,13 @@ private Expression removeDynamicFilters(Expression expression, Set<String> allow
         {
             return combineConjuncts(metadata, extractConjuncts(expression)
                     .stream()
+                    .map(predicate -> {
+                        // Remove all nested dynamic filters
+                        if (predicate instanceof LogicalBinaryExpression) {
```

Comment:
just use `io.prestosql.sql.tree.ExpressionTreeRewriter#defaultRewrite`. It doesn't change top-level node.

Reply:
I've tried to simplify this. If there's further scope for improvement here, please let me know.